### PR TITLE
Add Guest Dashboard API Endpoint 

### PR DESCRIPTION
**Pull request by agosmou**
## Problem
We need to start setting up the end points for the guest dashbard

## Solution
I added a skeleton for the guest dashboard endpoints, starting with:
- GET Tasks by guest id and application id

## Considerations
- I didn't push up any changes with the alembic migrations. I need to better understand how to work with SQLite
    - The `guest_dashboard_repository.py` has values hard-coded into it due to the above - I'll try to learn about how to use SQLite and get the database up and running in my environment so I can refactor this code.
    - due to the repository having hard coded values, other files are written with this consideration, so I will have to refactor throughout to consider real db data.
        - e.g. in `/api/openapi_server/test` the `populate_test_database_for_guest_dashboard` definition does not take a `num_entries` parameter from the `test_guest_dashboard_controller.py`.
        - e.g. `guest_dashboard_controller.py` also has hardcoded values

## Resources
- #500 
- https://github.com/hackforla/HomeUniteUs/issues/500#issuecomment-1648788279

### CODE REVIEW

**Comment by ju1es** [comment]
Description feedback:
- Problem statement, what issue is this addressing? I would be more pointed in your descriptions i.e "Guest Dashboard needs endpoint that fetches application steps."
- Solution statement, list more granular and tangible steps. i.e. added endpoint to specification, add tests, abstracted db logic into repository class.

File: api/openapi_server/controllers/guest_dashboard_controller.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+guest_dashboard_repository = GuestDashboardRepository()
+
+def get_guest_dashboard_by_id_and_application_id(guest_id,application_id):
+    """sumary_line
+    # noqa: E501
+    Keyword arguments:
+    argument -- description
+    Return: return_description
+    """
+    # tasks = guest_dashboard_repository.get_application_tasks(guest_id,application_id)
```
Comment (ju1es) [inline api/openapi_server/controllers/guest_dashboard_controller.py:18]: remove this if not used

File: api/openapi_server/controllers/guest_dashboard_controller.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    # if tasks != None:
+    #     return tasks, 200
+    # else:
+    #     return "tasks not found", 404
+
+
+    ################################ Testing OpenAPI UI ################################
+    tasks = [
```
Comment (ju1es) [inline api/openapi_server/controllers/guest_dashboard_controller.py:26]: we need to talk about this offline but we should think about if this is something that's hardcoded into the UI for now and we just pass back statuses. which means we'd need to fetch task statuses.

File: api/openapi_server/controllers/guest_dashboard_controller.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        },
+        {
+            "Host Selection": "Host Selection Complete"
+        },
+        {
+            "Match": "Match Complete"
+        }
+    ]
+
+    result = {
```
Comment (ju1es) [inline api/openapi_server/controllers/guest_dashboard_controller.py:38]: create an interface for this? I'd look into it but i imagine itd be something like below. The reason being if we have interfaces that define the expected structure being passed around it does a couple of things,
- minimizes programming typos/errors
- starts to formalize an agreement between callee and caller (i.e. front and back end)
- reads better (personally)

class TasksResponse:
    def __init__(self, guest_id, application_id, tasks):
       ...

File: api/openapi_server/models/database.py [hunk 4]
```diff
@@ -40,6 +40,14 @@ class Applicant(Base):
     applicant_status = Column(Integer, ForeignKey('applicant_status.id'), nullable=False)
     user = Column(Integer, ForeignKey('user.id'), nullable=False)
 
+class GuestApplications(Base):
+    __tablename__ = "guest_applications"
+
+    id = Column(Integer, primary_key=True, index=True)
```
Comment (ju1es) [inline api/openapi_server/models/database.py:46]: I think specifying primary_key is enough to create an index. `index` might be redundant here. not sure but just noting for future reviewers

File: api/openapi_server/openapi/paths/guestApplicationManagement.yaml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+get:
+  summary: Get a list of tasks to complete application.
+  tags:
+    - guest_dashboard
+  operationId: get_guest_dashboard_by_id_and_application_id
```
Comment (ju1es) [inline api/openapi_server/openapi/paths/guestApplicationManagement.yaml:5]: i would rename this. 'dashboard' can imply a lot of things but the summary is very specific about getting tasks for an application. maybe, 
get_guest_application_tasks

File: api/openapi_server/models/guest_dashboard_with_guest_id_and_application_id.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+from datetime import date, datetime  # noqa: F401
+
+from typing import List, Dict  # noqa: F401
+
+from openapi_server.models.base_model_ import Model
+from openapi_server import util
+
+
+
+class GuestDashboardWithGuestIdAndApplicationId(Model):
```
Comment (ju1es) [inline api/openapi_server/models/guest_dashboard_with_guest_id_and_application_id.py:13]: I would rename this and tailor to the 'get_guest_application_tasks' endpoint

File: api/openapi_server/openapi/schemas/_index.yaml [hunk 7]
```diff
@@ -27,25 +27,25 @@ ApplicantType:
 HousingProgram:
   type: object
   properties:
-    program_name:           
+    program_name:
       title: program_name
       type: string
     service_provider:
-      $ref: '#/ServiceProvider'
+      $ref: "#/ServiceProvider"
```
Comment (ju1es) [inline api/openapi_server/openapi/schemas/_index.yaml:34]: can you undo these changes? can move the linting/style check stuff into a separate PR.

File: api/openapi_server/openapi/schemas/_index.yaml [hunk 8]
```diff
@@ -96,11 +96,23 @@ IntakeQuestionType:
...
+    response_text:
       title: response_text
       type: string
     intake_question:
-      $ref: '#/IntakeQuestion'
+      $ref: "#/IntakeQuestion"
   required:
     - response_text
     - intake_question
+GuestDashboard:
```
Comment (ju1es) [inline api/openapi_server/openapi/schemas/_index.yaml:107]: keep this stuff though.

File: api/openapi_server/repositories/guest_dashboard_repository.py [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+        :param application_id: application id
+        :type application_id: int
+
+        :rtype: GuestDashboardWithGuestIdAndApplicationId
+        """
+        session = self._get_session()
+        row = self._generate_row(guest_dashboard)
+        session.add(row)
+        session.commit()
+        # guest_dashboard["id"] = row.id
```
Comment (ju1es) [inline api/openapi_server/repositories/guest_dashboard_repository.py:36]: remove this

**Comment by paulespinosa** [comment]
Hi @agosmou and @ju1es, how's it going on this PR? Can I help out with merging main into this branch?
